extensions/gitpod-web/src/portViewProvider.ts

```diff
@@ -0,0 +1,141 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import { GitpodExtensionContext, ExposedServedGitpodWorkspacePort, GitpodWorkspacePort, isExposedServedGitpodWorkspacePort, isExposedServedPort, PortInfo } from 'gitpod-shared';
+import { PortsStatus } from '@gitpod/supervisor-api-grpc/lib/status_pb';
+import { isUserOverrideSetting } from './experiments';
+
+const PortCommands = <const>['tunnelNetwork', 'tunnelHost', 'makePublic', 'makePrivate', 'preview', 'openBrowser', 'retryAutoExpose', 'urlCopy', 'queryPortData'];
+
+type PortCommand = typeof PortCommands[number];
+
+export class GitpodPortViewProvider implements vscode.WebviewViewProvider {
```

Comment:
I wonder why we cannot move everything to shared, and then pass options for customizations?

There seem to be still quite duplication between remote and web?